clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/streaming/ClientComputeRecordStreamDecoder.java

```diff
@@ -0,0 +1,78 @@
+package com.linkedin.venice.client.store.streaming;
+
+import com.linkedin.venice.HttpConstants;
+import com.linkedin.venice.client.exceptions.VeniceClientException;
+import com.linkedin.venice.client.store.transport.TransportClientStreamingCallback;
+import com.linkedin.venice.compression.CompressionStrategy;
+import com.linkedin.venice.read.protocol.response.streaming.StreamingFooterRecordV1;
+import com.linkedin.venice.serializer.RecordDeserializer;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Executor;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+
+public class ClientComputeRecordStreamDecoder<K, V> implements TransportClientStreamingCallback {
+  private final Function<Map<String, String>, TransportClientStreamingCallback> decoderProvider;
+  private TransportClientStreamingCallback decoder = null;
+
+  public ClientComputeRecordStreamDecoder(
+      List<K> keyList,
+      ClientComputeStreamingCallback<K, V> callback,
+      StreamingCallback trackingCallback,
+      Executor deserializationExecutor,
+      RecordDeserializer<StreamingFooterRecordV1> streamingFooterRecordDeserializer,
+      Supplier<RecordDeserializer<V>> computeDeserializerProvider,
+      Function<Integer, RecordDeserializer<V>> valueDeserializerProvider,
+      BiFunction<CompressionStrategy, ByteBuffer, ByteBuffer> decompressor) {
+    this.decoderProvider = headers -> {
+      if (headers.containsKey(HttpConstants.VENICE_CLIENT_COMPUTE)) {
+        callback.onRemoteComputeStateChange("0".equals(headers.get(HttpConstants.VENICE_CLIENT_COMPUTE)));
+        return new MultiGetRecordStreamDecoder<>(keyList, new StreamingCallback<K, V>() {
```

Comment:
I find this part a bit confusing. It appears that as long as the response has header "VENICE_CLIENT_COMPUTE", we expect the responses are multi-get results, regarding of the header values; but instead of always setting false in "onRemoteComputeStateChange", we check whether it's equal to "0".

Reply:
This is because only multi-get sets `VENICE_CLIENT_COMPUTE` header. When this header is used in the response, it indicates the recommended compute processing mode. This way clients who were doing client-compute using multi-get can detect a new recommendation to switch to remote computation when Venice service operator turns on read compute for the store.